src/senaite/core/content/sampletemplate.py

```diff
@@ -0,0 +1,635 @@
+                uid = api.get_uid(v.get("uid"))
+                hidden = v.get("hidden",
+                               api.get_object(uid).getHidden())
+                part_id = v.get("part_id", "")
+            elif api.is_object(v):
+                uid = api.get_uid(v)
+                hidden = v.getHidden()
+            elif api.is_uid(v):
+                obj = api.get_object(v)
+                uid = v
+                hidden = obj.getHidden()
+            else:
+                raise TypeError(
+                    "Expected object, uid or record, got %r" % type(v))
+            records.append({
+                "uid": uid,
+                "hidden": hidden,
+                "part_id": part_id,
+            })
+
+        mutator = self.mutator("services")
+        mutator(self, records)
+
+    # BBB: AT schema field property
+    Services = property(getServices, setServices)
+
+    @security.protected(permissions.View)
+    def getAnalysisServicesSettings(self):
+        """BBB: Return the settings for all assigned services
+
+        :returns: List of dicts including `uid`, `hidden` and `part_id`
+        """
+        # Note: We store the selected service UIDs, hidden and part_id
+        # settings in the `services` field. Therefore, we can just return the
+        # raw value.
+        return self.getRawServices()
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setAnalysisServicesSettings(self, settings):
+        """BBB: Update settings for all assigned service UIDs
+
+        This method expects a list of dictionaries containing the service
+        `uid`, `part_id` and the `hidden` setting.
+
+        This is basically the same format as stored in the `services` field!
+
+        However, we want to just update the settings for selected service UIDs"
+
+        >>> settings =  [{'uid': '...', 'hidden': False, 'part_id': 'part-1'}]
+        >>> setAnalysisServicesSettings(settings)
+        """
+        if not isinstance(settings, list):
+            settings = [settings]
+
+        by_uid = self.get_services_by_uid()
+
+        for setting in settings:
+            if not isinstance(setting, dict):
+                raise TypeError(
+                    "Expected a record containing `uid`, `hidden` and"
+                    "`partition`, got %s" % type(setting))
+            uid = api.get_uid(setting.get("uid"))
+            hidden = setting.get("hidden", api.get_object(uid).getHidden())
+            part_id = setting.get("part_id", "")
+
+            if not uid:
+                raise ValueError("UID is missing in setting %r" % setting)
+
+            record = by_uid.get(api.get_uid(uid))
+            if not record:
+                continue
+            record["hidden"] = hidden
+            record["part_id"] = part_id
+
+        # set back the new services
+        self.setServices(by_uid.values())
+
+    @security.protected(permissions.View)
+    def getAnalysisServiceSettings(self, service):
```

Comment:
Noted. For the moment I leave it as it is, because I can not overlook the side-effects on some other parts. Also it is not a critical method. Therefore, I do not want to raise and error if not found.